pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -659,29 +661,74 @@ type InterfaceTarget struct {
 }
 
 type Alias struct {
+	name        string
+	userDefined bool
+}
+
+// Package private, responsible to interact with xml.Encoder and xml.Decoder
+type userAliasXML struct {
 	Name string `xml:"name,attr"`
 }
 
-type UserAlias Alias
+// Package private, responsible to interact with json.Marshal and json.Unmarshal
+type userAliasJSON struct {
+	Name        string
+	UserDefined bool
+}
 
 type Rom struct {
 	Enabled string `xml:"enabled,attr"`
 }
 
+func NewLibvirtManagedAlias(aliasName string) *Alias {
+	return &Alias{name: aliasName}
+}
```

Comment:
this function is never used is it? why do we need this?

Reply:
The function is only used the functional test.

Reply:
@davidvossel I would propose we can test this. We just need to create a VM where are least some of the device names were auto-generated (not named by kubevirt code, so lack the "ua-" prefix). then check that the names are identical before and after migration.

The reason this is a valid test is because currently we're actually mis-handling names in this case and just happen to be lucky that nothing collides. Fixing that proves we've fixed the device trimming error.

Reply:
@yuhaohaoyu I verified a few migration scenarios and could create a situation where your bug appears. I can confirm that it fixes your problem.

Would you mind modifying in a followup PR one of the migration e2e tests so that the VMI which migrates has a table device defined?

For instance, this VMI will fail to migrate without your changes:

```yaml
---
apiVersion: kubevirt.io/v1
kind: VirtualMachineInstance
metadata:
  labels:
    special: vmi-nocloud
  name: vmi-nocloud
spec:
  domain:
    devices:
      inputs:
      - bus: usb
        name: tablet
        type: tablet
      interfaces:
      - masquerade: {}
        name: default
      disks:
      - disk:
          bus: virtio
        name: containerdisk
      - disk:
          bus: virtio
        name: cloudinitdisk
      - disk:
          bus: virtio
        name: emptydisk
    machine:
      type: ""
    resources:
      requests:
        memory: 64M
  terminationGracePeriodSeconds: 0
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/cirros-container-disk-demo:devel
    name: containerdisk
  - cloudInitNoCloud:
      userData: |
        #!/bin/sh

        echo 'printed from cloud-init userdata'
    name: cloudinitdisk
  - emptyDisk:
      capacity: 2Gi
    name: emptydisk
  networks:
  - name: default
    pod: {}
```

Note this section:


```yaml
      inputs:
      - bus: usb
        name: tablet
        type: tablet
```

Just adding such a device to a simple existing migration test should help us already enough.

Reply:
Will do this right away.